rust/parser/src/lexer.rs

```diff
@@ -1669,6 +1669,13 @@ impl<'alloc> Lexer<'alloc> {
                         self.chars.next();
                         return Ok((SourceLocation::new(start, self.offset()), None, TerminalId::Coalesce));
                     }
+                    Some('.') => {
+                        self.chars.next();
+                        if let Some('0'..='9') = self.peek() {
+                            return Ok((SourceLocation::new(start, self.offset()), None, TerminalId::QuestionMark))
```

Comment:
I don't think this works, because the `.` was consumed by `self.chars.next()`. This would cause `true?.9:0` to give the result `9`.

I think `self.peek()` can't do the job, because it requires 2-character lookahead. But look at how `self.peek()` is implemented; the same idea works for peeking 2 characters ahead instead of 1.

Reply:
aahh i see, it was dropping the `.` making the number `3`. that was definitely wrong. I fixed it... now its just the name im not sure about